RULES/hdl/xsim.go

```diff
@@ -258,59 +322,86 @@ func xsimVerbosityLevelToFlag(level string) (string, bool) {
 	return verbosity_flag, print_output
 }
 
-// xsimCmd will create a command for starting 'vsim' on the compiled and optimized design with flags
+// xsimCmd will create a command for starting 'xsim' on the compiled and optimized design with flags
 // set in accordance with what is specified on the command line.
 func xsimCmd(rule Simulation, args []string, gui bool, testcase string, params string) string {
-	// Prefix the vsim command with this
+	// Prefix the xsim command with this
 	cmd_preamble := ""
 
-	// Default log file
-	log_file_suffix := "xsim.log"
+	// Default log and wdb file names
+	file_suffix := ""
+	file_spacer := ""
 	if testcase != "" {
-		log_file_suffix = testcase + "_" + log_file_suffix
+		file_suffix = testcase
+		file_spacer = "_"
 	}
 	if params != "" {
-		log_file_suffix = params + "_" + log_file_suffix
+		if file_suffix != "" {
+			file_suffix = "_" + file_suffix
+		}
+		file_suffix = params + file_suffix
+		file_spacer = "_"
 	}
-	log_file := rule.Path().WithSuffix("/" + log_file_suffix)
 
-	// Default flag values
-	vsim_flags := " --onfinish quit --log " + log_file.String()
-	seed_flag := " --sv_seed 1"
-	verbosity_flag := " --testplusarg verbosity=DVM_VERB_NONE"
-	mode_flag := ""
-	plusargs_flag := ""
+	log_file := rule.Path().WithSuffix("/" + file_suffix + file_spacer + "xsim.log")
+	wdb_file := rule.Path().WithSuffix("/" + file_suffix + ".wdb")
 
-	// Collect do-files here
-	var do_flags []string
+	// Script to execute
+	do_file := rule.Path().WithSuffix("/" + "xsim.tcl")
 
-	// Turn off output unless verbosity is activated
-	print_output := false
+	// Default flag values
+	seed := int64(1)
+	xsim_cmd := []string{
+		"xsim",
+		"--log", log_file.String(),
+		"--wdb", wdb_file.String(),
```

Comment:
(?) So there's the new flag now to enable dumping to wdb. 
In the `tcl_file_template` you enable logging if the flag is set.
But it seems the wdb file is created regardless of the flag.
I presume it will just be empty?

Reply:
Fixed.